scl/cri/cri.conf

```diff
@@ -0,0 +1,51 @@
+#########################################################
+# Parser for CRI log format
+# ISO_TIME(nano) SP STREAM SP TAG SP MSG
+# https://github.com/kubernetes/kubernetes/issues/44976
+#========================================================
+# Retrieves information about K8S meta from log filename:
+#  - container ID
+#  - container Name
+#  - POD NameSpace
+#  - POD Name
+# Use .cri prefix
+# Set .cri.time as S_ISODATE with use of date-parser()
+#
+# Note: does not support multiline, line = message
+#########################################################
+# Example log:
+# 2022-02-04T18:14:43.219493781+01:00 stdout F Starting up on port 80
+
+@define crio-log-regexp "([a-z0-9-]+)_([a-z0-9-]+)_([a-z0-9]+)-([a-z0-9]{64}).log"
+
+block rewrite extract-container-id (value("MESSAGE")){
+    # make the container-id of 12 characters long as usual in cli
+    subst("`crio-log-regexp`", "$(substr $4 0 12)", value("`value`"), flags(global, store-matches), type(pcre));
+};
+
+block parser cri-parser(prefix('.cri.')) {
+    channel {
+        parser {
+            csv-parser(
+                columns("`prefix`time", "`prefix`stream", "`prefix`flag", "`prefix`message")
+                delimiters(" ")
+                flags(greedy)
+            );
+            # set S_ISODATE to value of ${.cri.time}
+            date-parser(format("%FT%H:%M:%S.%f%Z") template("${.cri.time}"));
+        };
+	# drop everything which does not match (to exclude logs)
+        #filter { not match("" value("`prefix`COMMAND") type(string)); };
+        rewrite {
+            set("$(basename \"${FILE_NAME}\")", value("`prefix`container.id"));
+            extract-container-id(value("`prefix`container.id"));
+            set("$1", value("`prefix`pod.name"));
+            set("$2", value("`prefix`pod.namespace"));
+            set("$3", value("`prefix`container.name"));
+        };
+    };
+};
+
+application cri[syslog] {
+    parser { cri-parser(); };
```

Comment:
Can you add a filter that pre-classifies this log format?
For performance reason it's much better to find a filter expression that exclude irrelevant logs

Reply:
The only meaningful filter can be something like:
"FILE_NAME=/var/log/containers/"
other filters for message processing might be set in syslog-ng configuration. Some K8s namespace or POD might be out of interest.

Following are all variables available prior parsing (file source with no-parse flag set):
FILE_NAME=/var/log/containers/coredns-84c56f7bfb-rjb99_kube-system_coredns-5fa6c8559123e406fcfa49bf60a01908498d34480d811720d145299a166e8b04.log
HOST=s_containers@syslog-xp8lx
HOST_FROM=syslog-xp8lx
MESSAGE="2022-02-03T10:03:52.545549629+01:00 stdout F [WARNING] No files matching import glob pattern: /etc/coredns/custom/*.server"
SOURCE=s_containers